src/SignalR/clients/csharp/Client/src/HubConnectionBuilderHttpExtensions.cs

```diff
@@ -138,8 +141,20 @@ private static IHubConnectionBuilder WithUrlCore(this IHubConnectionBuilder hubC
                 hubConnectionBuilder.Services.Configure(configureHttpConnection);
             }
 
+            // Add HttpConnectionOptionsDerivedHttpEndPoint so HubConnection can read the Url from HttpConnectionOptions
+            // without the Signal.Client.Core project taking a new dependency on Http.Connections.Client.
+            hubConnectionBuilder.Services.AddSingleton<EndPoint, HttpConnectionOptionsDerivedHttpEndPoint>();
+
             hubConnectionBuilder.Services.AddSingleton<IConnectionFactory, HttpConnectionFactory>();
             return hubConnectionBuilder;
         }
+
+        private class HttpConnectionOptionsDerivedHttpEndPoint : HttpEndPoint
```

Comment:
This name is interesting... maybe `HttpEndPointWithConnectionOptions`?

Reply:
Meh, it's a private class. That's why I went with the overly-descriptive name.

Reply:
> That's why I went with the overly-descriptive name.

False, you do this in general 😄